app/views/courses/TeacherClassView.coffee

```diff
@@ -329,8 +329,10 @@ module.exports = class TeacherClassView extends RootView
       courseOrder.push(course.id)
     csvContent = "data:text/csv;charset=utf-8,Username,Email,Total Playtime,#{courseLabels}Concepts\n"
     levelCourseMap = {}
+    language = @classroom.get('aceConfig')?.language
     for trimCourse in @classroom.get('courses')
       for trimLevel in trimCourse.levels
+        continue if language and trimLevel.primerLanguage is language
```

Comment:
(continued from app/models/Classroom.coffee)
And then here between the two loops, make a Levels collection out of `trimCourse.levels`, and call that filtering function on it.